File: chapter5/java/designpattern/src/main/java/com/designpattern/PepperoniPizza.java
package com.designpattern;

/**
 * PepperoniPizza is a concrete class extending abstract class Pizza. It implements abstract methods as per the requirement to prepare a Pepperoni pizza.
 *
 * @Author Bridget Wu
 */
public class PepperoniPizza extends Pizza {
    /**
     * add Pepperoni topping
     */
    @Override
    void addTopping() {
        System.out.println("=== Add Pepperoni topping === ");
    }
}

File: chapter5/typescript/PepperoniPizza.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PepperoniPizza = void 0;
const Pizza_1 = require("./Pizza");
/**
 * PepperoniPizza is a concrete class extending abstract class Pizza. It implements abstract methods as per the requirement to prepare a Pepperoni pizza.
 *
 * @Author Bridget Wu
 */
class PepperoniPizza extends Pizza_1.Pizza {
    /**
     * add Pepperoni topping
     */
    addTopping() {
        console.log("=== Add Pepperoni topping === ");
    }
}
exports.PepperoniPizza = PepperoniPizza;
